packages/docusaurus-theme-redoc/src/theme/Redoc/ServerRedoc.tsx

```diff
@@ -45,4 +48,23 @@ function ServerRedoc(
   );
 }
 
+function collectMenuItemAnchors(menuItems: IMenuItem[], parentAnchor = "") {
+  menuItems.forEach((menuItem) => {
+    // Register anchor for menu item
+    useBrokenLinks().collectAnchor(menuItem.id);
```

Comment:
doesn't this break rules of hooks?

Reply:
Thanks!  Was not aware of the guidance around calling hooks, just stopped once I got something working.  I've updated to memoize the broken link checker, along the lines of the suggestion in your other comment.